### PR TITLE
Bump version and update documentation

### PR DESCRIPTION
**Pull request by jafyvilla**
Release preparations. Once the review is finished I can tag it and push the generated documentation to bluebrain.github.com

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -22,9 +22,9 @@ if(CMAKE_SYSTEM_NAME MATCHES "Linux")
 endif()
 
 set(VERSION_MAJOR "0")
-set(VERSION_MINOR "3")
+set(VERSION_MINOR "4")
```
Comment (tribal-tec) [inline CMakeLists.txt]: 3

File: CMakeLists.txt [hunk 2]
```diff
@@ -22,9 +22,9 @@ if(CMAKE_SYSTEM_NAME MATCHES "Linux")
...
 set(VERSION_PATCH "0")
-set(VERSION_ABI 2) # OFF or ABI version
+set(VERSION_ABI 3) # OFF or ABI version
```
Comment (tribal-tec) [inline CMakeLists.txt]: 2?

File: README.md [hunk 3]
```diff
@@ -1,6 +1,39 @@
...
+* [LIV-157](https://bbpteam.epfl.ch/project/issues/browse/LIV-157):
+  Livre renders overlapping LOD nodes from different levels
+
+## About
+
+The following platforms and build environments are tested:
+
+* Linux: Ubuntu 14.04, RHEL 6.5 (Makefile, x64)
+
+The [API documentation](http://bluebrain.github.io/Livre-0.4/index.html)
```
Comment (tribal-tec) [inline README.md]: 0.3

File: doc/Changelog.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
+# Changelog {#Changelog}
+
+# Release 0.4 {#Release040}
+
+* Add command line option to enable ZeroEQ camera synchronization
```
Comment (tribal-tec) [inline doc/Changelog.md]: link to pull requests if possible/feasible?

File: README.md [hunk 5]
```diff
@@ -1,6 +1,39 @@
-[![Build Status](https://travis-ci.org/BlueBrain/Livre.png?branch=master)](https://travis-ci.org/BlueBrain/Livre)
+# Livre
+![Livre](doc/images/livre_small.png)
```
Comment (eile) [inline README.md]: Can we update this?

File: README.md [hunk 6]
```diff
@@ -1,6 +1,39 @@
...
+* Visualisation of pre-processed UVF format
+  ([source code](https://github.com/SCIInstitute/Tuvok.git)) volume data sets.
+* Real-time voxelisation of different data sources (surface meshes, BBP morphologies,
+  local-field potentials, etc) through the use of plugins.
+* Multi-node, multi-gpu rendering (Currently only sort-first rendering)
+
+
+To keep track of the changes between releases check the [changelog](doc/Changelog.md).
+
+Contact: ahmet.bilgili@epfl.ch
```
Comment (eile) [inline README.md]: bbp-open-source@googlegroups.com

File: doc/Changelog.md (hunk 4 above)
Comment (eile) [inline doc/Changelog.md]: s/Add/New/g

File: doc/Changelog.md [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+* Add command line parameter to set the camera lookAt vector to define its orientation
+* Add command line parameters for RESTBridge (hostname, port, ZeroEQ schema)
+* Add command line parameters to specify the minimum and maximum levels of detail to render
+* Add frustum culling
```
Comment (eile) [inline doc/Changelog.md]: Improved rendering performance through frustum culling

File: doc/Changelog.md [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+* Add livre_batch.py script for offline image series rendering
+* Add the possibility of specifying the rage of frames to render from the command line
```
Comment (eile) [inline doc/Changelog.md]: Rage? Was it that hard? :)

New command line option to specify ...

File: doc/Changelog.md [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+* Add time support, implementing a new animation mode that triggers a new frame
+  after the rendering of the previous one is finished
+* Do not setup ZeroEQ subscribers for REST if not requested
```
Comment (eile) [inline doc/Changelog.md]: Fix: do not...

File: doc/Changelog.md [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+* Fix a problem that made the texture cache not to unload unused elements
+* Fix the "Empty servus implementation" exception that was raised when zeroconf wasn not available
+* Fix the LOD computation based on the screen height
+* Frames can now be written to disk as PNG files
```
Comment (eile) [inline doc/Changelog.md]: Move up. Group by features, opt, fixes?

File: doc/Changelog.md [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+* General cleanup of command line parameters
+* General code cleanups and improvements
```
Comment (eile) [inline doc/Changelog.md]: Don't care. each release (should) have this.

File: doc/Changelog.md [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+* Graceful exit when ZeroEQ EXIT event is received
+* Refactor the code to setup the data and texture UploadProcessors, removing race
+  conditions and fixing crashes (at exit and with multiple local GPUs) in the process
+* Several fixes in the tree creation and the LOD-selection algorithm
+* Synchronous mode (wait until every block has been processed before rendering them)
+  is now available as a command line option
```
Comment (eile) [inline doc/Changelog.md]: Feature. move up.

File: doc/Changelog.md [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+* The hierarchical information for the Dash tree is now implicit in each node ID,
+  effectively becoming a flat structure. This change has a positive impact on the
+  performance and memory use
```
Comment (eile) [inline doc/Changelog.md]: Don't care about the detail.

Improved performance and memory usage through a new implicit, flat octtree implementation.

File: doc/Changelog.md [hunk 14]
```diff
@@ -0,0 +1,31 @@
...
+* Qt5 replaces Qt4 dependency (used for the GUI in the transfer function editor)
```
Comment (eile) [inline doc/Changelog.md]: Upgrade to Qt5 for the transfer function editor

**Comment by jafyvilla** [comment]
Updated

File: README.md (hunk 5 above)
Comment (jafyvilla) [inline README.md]: The image? It's already different from the one we had. Shall I use something else?

Comment (eile) [inline README.md]: Yes please. Something which shows something.

File: doc/Changelog.md [hunk 15]
```diff
@@ -0,0 +1,50 @@
+# Changelog {#Changelog}
+
+# Release 0.3 (2015-07-07){#Release030}
```
Comment (eile) [inline doc/Changelog.md]: ##

File: doc/technical-overview.dox [hunk 16]
```diff
@@ -58,28 +57,24 @@ the required nodes.
...
 and Rendering.
-- Eq : Collection of implementations of classes for Equalizer classes.
-- bbpsdkvox : Collection of implementations of classes for morphology voxelization.
-- fivox : Collection of implementations of classes for local-field potential voxelisation and
-visualization.
-- meshvox : Collection of implementations of classes for mesh voxelisation and visualization.
+- Eq: Collection of implementations of classes for Equalizer classes.
 
-And, additionally, there is the UVF library for supporting pre-processed UVF volume data sets.
+Additionally, there is the UVF library to support pre-processed UVF volume data sets.
```
Comment (eile) [inline doc/technical-overview.dox]: List by data sources: memory, uvf, remote.

**Comment by jafyvilla** [comment]
Done

**Comment by jafyvilla** [comment]
Retest this please

**Comment by tribal-tec** [comment]
+1

**Comment by jafyvilla** [comment]
Added changes from gerrit

File: README.md (hunk 5 above)
Comment (eile) [inline README.md]: New one looks much better. Can you also upload the large version?

File: doc/technical-overview.dox [hunk 17]
```diff
@@ -58,28 +57,28 @@ the required nodes.
...
+- Lib: Collection of implementations of classes for Cache such as "Data Cache", "Texture Cache"
 and Rendering.
-- Eq : Collection of implementations of classes for Equalizer classes.
-- bbpsdkvox : Collection of implementations of classes for morphology voxelization.
-- fivox : Collection of implementations of classes for local-field potential voxelisation and
-visualization.
-- meshvox : Collection of implementations of classes for mesh voxelisation and visualization.
 
-And, additionally, there is the UVF library for supporting pre-processed UVF volume data sets.
+Additionally, there are 3 differente data sources that make use of the VolumeDataSourcePlugin
```
Comment (eile) [inline doc/technical-overview.dox]: differente, si si si! ;)

File: doc/technical-overview.dox [hunk 18]
```diff
@@ -58,28 +57,28 @@ the required nodes.
...
+to support the following types of data sets:
+- remote: Proxy for a remote ZeroEQ-based data source service.
+- UVF: Supports pre-processed UVF volume data sets.
+- memory (livre/Lib/Data/MemoryDataSource): Generates random, in-memory volume data.
```
Comment (eile) [inline doc/technical-overview.dox]: Afaik not random.

File: README.md (hunk 5 above)
Comment (jafyvilla) [inline README.md]: Added now to illustrate the use of external data sources.

**Comment by eile** [comment]
+1

**Comment by jafyvilla** [comment]
Retest this please

**Pull request merged**